packages/schematics/angular/guard/schema.json

```diff
@@ -50,8 +50,12 @@
         "enum": ["CanActivate", "CanActivateChild", "CanDeactivate", "CanLoad", "CanMatch"],
         "type": "string"
       },
-      "default": ["CanActivate"],
       "x-prompt": "Which interfaces would you like to implement?"
```

Comment:
I think the prompt in this case should be removed. Since otherwise there wouldn't be a way to use `guardType`.

Reply:
I'm assuming there's not a way to make the prompt conditional? I feel like the lack of a prompt is a step down in UX.

Reply:
Correct. we don't support conditional prompts.

Reply:
So I tried this out and I don't think this is going to work. The experience now is that `ng g guard myGuard` will just generate a `canActivate` guard automatically. Most users will not know about the `implements` flag and will not know how to generate a guard of another type. We'll have to come up with another way to make this work.

Reply:
Okay, I updated the approach to use the `--functional` flag and throw an error if there is more than 1 value for `implements` when using it. The names of the existing `implements` options are the interfaces, and while this isn't ideal, I don't think it's really an issue.

PTAL